### ISSUE TITLE
Auth with VK.com is failed (remanga responses 400 error)

### ISSUE THREAD

**Issue by ilyasemenow90** [XITRIX/ReManga-iOS#6]
Привет. Спасибо за такой куртой проект!

Перестала работать авторизация через Вконтакте.

iPhone 12 Pro Max, iOS 17.2.1, XCode 15.2

Steps to reproduce:
1) открыть приложение, перейти на страницу логина
2) Выбрать вконтнакте
3) На открывшейся веб странице ВК залогиниться
4) нажать "Войти через VK ID"

Я подебажил немного и в консоли пишет вот такую ошибку 

>  0x11b044340 - [pageProxyID=6, webPageID=27, PID=22841, navigationID=10] ProvisionalPageProxy::didFailProvisionalLoadForFrame: frameID=1
0x15d04c618 - [pageProxyID=6, webPageID=7, PID=22834] WebPageProxy::didFailProvisionalLoadForFrame: frameID=1, isMainFrame=1, domain=WebKitErrorDomain, code=102, isMainFrame=1, willInternallyHandleFailure=0
Error acquiring assertion: <Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}>
0x11b020a80 - ProcessAssertion::acquireSync Failed to acquire RBS assertion 'XPCConnectionTerminationWatchdog' for process with PID=22841, error: Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}

В веб инспекторе в консоли есть только вот такая ошибка: 
`Failed to load resource: the server responded with a status of 404 () https://static.vk.com/vkid/1.1.878/auth.js.map`

Попробовал пару "фиксов" из интернета (добавить background modes, разрешить previewLinks `webView.allowsLinkPreview = true`, добавить аргумент для запуска `OS_ACTIVITY_MODE = disable`) ничего не помогло.

Может ли быть такое что сломалось только у меня одного? Есть ли какие-то идеи как это можно было бы исправить?

Небольшое видео как это происходит 

https://github.com/XITRIX/ReManga-iOS/assets/765053/23ca66dc-9ac0-4ddf-bc1b-468aef4af554

**Comment by ilyasemenow90** [comment]
Так, после небольшого дебага, стала понятно что проблема не в vk.com, а в том, что запрос к реманге на получение токена фейлится с 400 ошибкой
Вот запрос и ответ:
`url: https://api.remanga.org/api/users/social/`

```
{
  "code": "", << Мой код от ВК тут
  "provider": "vk",
  "referral": "",
  "redirect_uri": "https://remanga.org/social"
}
```

response:
```
{
  "msg": "User authorization failed: invalid access_token (4)."
}
```

<img width="1169" alt="Screenshot 2024-05-05 at 18 40 09" src="https://github.com/XITRIX/ReManga-iOS/assets/765053/ae011be9-f795-4540-af3a-39be67b4adbb">

**Comment by XITRIX** [comment]
Спасибо за детали, я попробую глянуть в чём проблема

**Comment by XITRIX** [comment]
Похоже что они переделали авторизацию через ВК и отключили старое API, новое какое то более мудрёное, вероятно надо добавить авторизацию через логин\пароль ...

**Comment by XITRIX** [comment]
Логин по почте добавил, пока не починю VK хоть как то да можно будет зайти

**Comment by ilyasemenow90** [comment]
Спасибо! У меня вариант с `.alert {` выдаёт ошибку при компиляции: `Missing arguments for parameters 'message', 'actions' in call`

Но если поменять на вот такой вариант, то работает.
```
            .init(title: "Почта", style: .default, action: {
                let alert = UIAlertController(title: "Авторизация", message: nil, preferredStyle: .alert)
                alert.addTextField { textFielt in
                    textFielt.placeholder = "Почта"
                    textFielt.textContentType = .username
                }
                alert.addTextField { textFielt in
                    textFielt.placeholder = "Пароль"
                    textFielt.textContentType = .password
                    textFielt.isSecureTextEntry = true
                }

                alert.addAction(.init(title: "Закрыть", style: .cancel))
                alert.addAction(.init(title: "Продолжить", style: .default, handler: { [unowned self] _ in
                    guard let login = alert.textFields?[0].text,
                          let password = alert.textFields?[1].text
                    else { return }

                    Task {
                        try await process(login: login, password: password, from: vm)
                    }

                }))

                vm.navigationService.present(alert, animated: true)
            }),
```
